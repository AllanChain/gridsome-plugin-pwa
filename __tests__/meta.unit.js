const { defaultOptions } = require('../gridsome.server')
const defaultsDeep = require('lodash/defaultsDeep')
const client = require('../gridsome.client')

const clientHead = // in js, no such default param behavior like python
  (userOptions, head = { head: { meta: [], link: [] } }) => {
    const options = {
      name: 'Awesome Gridsome',
      publicPath: '/gridsome/',
      msTileImage: '/gridsome/assets/icons/favicon.png-144x144.png',
      ...defaultsDeep(userOptions, defaultOptions())
    }
    client(null, options, head)
    return head
  }

it('works with zero config', () => {
  expect(JSON.stringify(clientHead({}))).toMatchSnapshot()
})
