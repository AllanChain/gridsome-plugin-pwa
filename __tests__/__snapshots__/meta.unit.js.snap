// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with zero config 1`] = `"{\\"head\\":{\\"meta\\":[{\\"name\\":\\"theme-color\\",\\"content\\":\\"#00a672\\"},{\\"name\\":\\"apple-mobile-web-app-capable\\",\\"content\\":\\"no\\"},{\\"name\\":\\"apple-mobile-web-app-status-bar-style\\",\\"content\\":\\"default\\"},{\\"name\\":\\"apple-mobile-web-app-title\\",\\"content\\":\\"Awesome Gridsome\\"},{\\"name\\":\\"application-name\\",\\"content\\":\\"Awesome Gridsome\\"},{\\"name\\":\\"msapplication-TileColor\\",\\"content\\":\\"#00a672\\"},{\\"name\\":\\"msapplication-TileImage\\",\\"content\\":\\"/gridsome/assets/icons/msapplication-icon-144x144.png\\"}],\\"link\\":[{\\"rel\\":\\"manifest\\",\\"href\\":\\"/gridsome/manifest.json\\"},{\\"rel\\":\\"mask-icon\\",\\"href\\":\\"/gridsome/safari-pinned-tab.svg\\",\\"color\\":\\"#00a672\\"}]}}"`;
